### PR TITLE
479 - Authorized any connected user to comment open deal

### PR DESCRIPTION
**Pull request by giladt**
## What was done
Allowed any connected wallet to begin a discussion and/or comment on a thread.

### CODE REVIEW

File: src/dealDashboard/dealDashboard.ts [hunk 1]
```diff
@@ -27,11 +27,16 @@ export class DealDashboard {
...
-        ...this.deal.registrationData.partnerDAO?.representatives.map(representative => representative.address) || "",
-      ].includes(this.ethereumService.defaultAccountAddress)
+      (
+        /* Open deal can not be private, and anyone can comment and begin a discussion */
+        !this.deal.registrationData.partnerDAO
+        /* In a Partnered deal only members can comment and begin a discussion */
+        || [
+          this.deal.registrationData.proposalLead?.address,
+          ...this.deal.registrationData.primaryDAO?.representatives.map(representative => representative.address) || "",
+          ...this.deal.registrationData.partnerDAO?.representatives.map(representative => representative.address) || "",
```
Comment (hiaux0) [inline src/dealDashboard/dealDashboard.ts:37]: For the future: 
Could maybe reuse methods from current PR from @IonelLupu 
https://github.com/PrimeDAO/prime-deals-dapp/pull/499/files#diff-73772712224a2110ee6dd67f201fb8721e0ea04e41a2f67bc214cc7ac088d059=

**Review by hiaux0 — COMMENTED** [review]
.

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/dealDashboard/dealDashboard.ts [hunk 2]
```diff
@@ -29,15 +29,10 @@ export class DealDashboard {
...
         /* Open deal can not be private, and anyone can comment and begin a discussion */
-        !this.deal.registrationData.partnerDAO
+        this.deal.isOpenProposal ||
         /* In a Partnered deal only members can comment and begin a discussion */
-        || [
-          this.deal.registrationData.proposalLead?.address,
-          ...this.deal.registrationData.primaryDAO?.representatives.map(representative => representative.address) || "",
-          ...this.deal.registrationData.partnerDAO?.representatives.map(representative => representative.address) || "",
-        ].includes(this.ethereumService.defaultAccountAddress)
+        [...this.deal.representativesAndLead].includes(this.ethereumService.defaultAccountAddress)
```
Comment (dkent600) [inline src/dealDashboard/dealDashboard.ts:34]: First, note the correct say to use Set.  But you don't need this anyway, use `deal.isRepresentativeUser` (which does it corrrectly)

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/dealDashboard/dealDashboard.ts [hunk 3]
```diff
@@ -22,4 +22,17 @@ export class DealDashboard {
     this.deal = this.dealService.deals.get(this.dealId);
     await this.deal.ensureInitialized();
   }
+
+  @computedFrom("ethereumService.defaultAccountAddress", "deal.registrationData")
+  get authorized(): boolean {
```
[the remark this answers is not in the document]
Comment (giladt) [reply src/dealDashboard/dealDashboard.ts:27]: this getter is actually not needed anymore, as it is not used anywhere- replaced in the view with deal properties.

**Review by hiaux0 — COMMENTED** [review]
ez fix

**Review by dkent600 — APPROVED** [review]

**Pull request merged**